src/types/common.ts

```diff
@@ -39,7 +45,13 @@ export type OptionalArweave<T = NonNullable<unknown>> = {
 export type OptionalPaymentUrl<T = NonNullable<unknown>> = {
   paymentUrl?: string;
 } & T;
-export type TurboArNSSigner = Signer;
+// TODO: TurboArNSSigner could be simply `Signer` but we need to implement each message signing method for signing headers.
+export type TurboArNSSigner =
+  | EthereumSigner
+  | InjectedEthereumSigner
+  | ArweaveSigner
+  | ArconnectSigner
+  | Window['arweaveWallet'];
```

Comment:
I'm considered we are not sufficiently abstracting signers. Ideally any `Signer` used satisfies the necesssary interface we determine based on the integration points with AO and Turbo. It already seems confusing to me you could have a `TurboArNSSigner` which is not satisfied by an `AoSigner` or vice versa

Reply:
how do we get unit tests for each signer type? what kind of life that would be?

Reply:
I agree - the issue we are facing here is we made the decision to not use data items for auth and instead sign messages - the tradeoff there is we need to access specific methods, like signMessage on arweaveWallet api, to be able to execute that. It also means we need to implement the api on each specific signer type. Might be able to be a bit looser here and use ContractSigner to a degree - can't use AoSigner because that one is _specifically_ for signing data items, but I think we can abstract it to Signer and Window['arweaveWallet'] atleast.

Reply:
I think I _can_ use Signer but its a bit wierd to allow all SIgners when in the headers signing we throw if its not an arweave or eth signature type

Reply:
node signers are easy to unit test - the web ones a little less so. Injected provider can probably be unit tested with an ethers non web provider, but arconnect gets tricky, needs a mock.

Reply:
do you intend on adding tests?